### ISSUE TITLE
nil-compare: support EqualValues with typed nil

### ISSUE THREAD

**Issue by Antonboom** [Antonboom/testifylint#222]
https://github.com/search?q=%2F%5C.EqualValues%5C%28t%2C.*%5C%28nil%5C%29%2C+%2F+language%3AGo+&type=code

```go
require.EqualValues(t, []base.DiskFileNum(nil), recycler.LogNumsForTesting())
require.EqualValues(t, ([]MailserverTopic)(nil), topics)
assert.EqualValues(t, (*tls.Config)(nil), opts.TLSConfig())
...
```

+ testcase for `expected-actual`

P.S. Caused by https://github.com/Antonboom/testifylint/issues/88
